nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -165,28 +165,28 @@ def _get_ratio_defining_params(
         if self._mode in [CompressWeightsMode.INT8_SYM, CompressWeightsMode.INT8_ASYM]:
             return all_weight_params
 
-        if self._all_layers:
-            return list(filter(lambda wp: len(wp.reduction_axes) == 1, all_weight_params))
-
         ratio_defining_params = list(
             filter(
                 lambda wp: wp.node_with_weight.metatype in self._backend_entity.matmul_metatypes,
                 all_weight_params,
             )
         )
 
+        # The last MatMul layer is quantized to 4-bits if all_layers=True or if the layer is shared
```

Comment:
This comment looks wrong.
`if the layer is shared` means that it has the same precision as embedding, and it's 4bit only with `all_layers`